constraint-solver/src/indexed_constraint_system.rs

```diff
@@ -50,9 +50,24 @@ impl<T: FieldElement, V> IndexedConstraintSystem<T, V> {
     pub fn bus_interactions(&self) -> &[BusInteraction<QuadraticSymbolicExpression<T, V>>] {
         &self.constraint_system.bus_interactions
     }
+
+    pub fn iter(&self) -> impl Iterator<Item = &QuadraticSymbolicExpression<T, V>> {
```

Comment:
I think we should either implement `IntoIterator` or rename these functions. I think `IntoIterator` is not justified because the iterator does not capture all aspects of the constraint system (i.e. how things are arranged in bus interactions). So maybe rename this function to `expressions()`?